src/Sylius/Component/Core/spec/OrderProcessing/OrderShipmentProcessorSpec.php

```diff
@@ -44,19 +46,34 @@ function it_is_an_order_processor()
         $this->shouldImplement(OrderProcessorInterface::class);
     }
 
-    function it_creates_a_single_shipment_with_default_shipping_method_and_assigns_all_units_to_it(
+    function it_creates_a_single_shipment_with_default_shipping_method_and_assigns_all_units_to_it_when_shipping_is_required(
         DefaultShippingMethodResolverInterface $defaultShippingMethodResolver,
         FactoryInterface $shipmentFactory,
         OrderInterface $order,
         OrderItemUnitInterface $itemUnit1,
         OrderItemUnitInterface $itemUnit2,
         ShipmentInterface $shipment,
-        ShippingMethodInterface $defaultShippingMethod
+        ShippingMethodInterface $defaultShippingMethod,
+        OrderItemInterface $orderItem,
+        ProductVariant $productVariant,
+        Collection $orderItems,
+        \Iterator $iterator
     ) {
         $defaultShippingMethodResolver->getDefaultShippingMethod($shipment)->willReturn($defaultShippingMethod);
 
         $shipmentFactory->createNew()->willReturn($shipment);
 
+        $orderItem->getVariant()->willReturn($productVariant);
+
+        $productVariant->isShippingRequired()->willReturn(true);
+
+        $orderItems->getIterator()->willReturn($iterator);
+
+        $iterator->rewind()->shouldBeCalled();
+        $iterator->valid()->willReturn(true);
+        $iterator->current()->willReturn($orderItem);
+
+        $order->getItems()->willReturn($orderItems);
```

Comment:
It can be `$order->getItems()->willReturn(new \ArrayIterator([$orderItem->getWrappedObject()]))`. Then, the whole part:
```
$orderItems->getIterator()->willReturn($iterator);

$iterator->rewind()->shouldBeCalled();
$iterator->valid()->willReturn(true);
$iterator->current()->willReturn($orderItem);
```
can be removed.